DEPRECATIONS.md

```diff
@@ -0,0 +1,23 @@
+# Deprecated APIs and features
+
+This document contains a summary of Feast APIs and features that have been deprecated or are scheduled to be deprecated, starting with version 0.19.
+
+## Index
+
+The following table lists deprecated APIs and features, organized by release. Each item is linked to a section later in this guide that describes the reason for deprecation.
+
+
+| API or Feature                                                                       | Removed by   |
+| :----------------------------------------------------------------------------------- | :----------- |
+| [`table_ref` parameter in `BigQuerySource`](#table_ref-parameter-in-bigquerysource)  | 0.20         |
+| [positional arguments in `Entity`](#positional-arguments-in-entity)                  | 0.23         |
```

Comment:
I'd prefer two columns here - when was a field marked as deprecated, and when was it removed. It's easier for users to understand how long something was deprecated before removal, instead of thinking that it was just yanked without warning.